src/ralph/api/auth.py

```diff
@@ -108,6 +109,13 @@ def get_stored_credentials(auth_file: Path) -> ServerUsersCredentials:
     return ServerUsersCredentials.parse_file(auth_file)
 
 
+@cached(
+    TTLCache(maxsize=10, ttl=3600),
+    key=lambda credentials: (
+        credentials.username,
+        credentials.password,
+    ),
```

Comment:
As we will deal with a global cache and multiple threads, shouldn't we provide a lock instance?

```suggestion
    ),
    lock=threading.Lock,
```

Reply:
Unless I am missing something, I don't see the use for a lock here, as there are only read operations in `authenticated_user`